source/grid/grid_tools.cc

```diff
@@ -3756,6 +3756,79 @@ namespace GridTools
       }
   }
 
+
+  template <int dim, int spacedim>
+  void
+  assign_co_dimensional_manifold_indicators(
+    Triangulation<dim, spacedim> &            tria,
+    const std::function<types::manifold_id(
+      const std::set<types::manifold_id> &)> &disambiguation_function)
+  {
+    // Easy case first:
+    if (dim == 1)
+      return;
+    const unsigned int size =
+      dim == 2 ? tria.n_lines() : tria.n_lines() + tria.n_quads();
+
+    // If user index is zero, then it has not been set.
+    std::vector<std::set<types::manifold_id>> manifold_ids(size + 1);
+    std::vector<unsigned int>                 backup;
+    tria.save_user_indices(backup);
+
+    unsigned next_index = 1;
+    for (auto cell : tria.active_cell_iterators())
+      {
+        if (dim > 1)
+          for (unsigned int l = 0; l < GeometryInfo<dim>::lines_per_cell; ++l)
+            {
+              if (cell->line(l)->user_index() == 0)
```

Comment:
I think you have to clear all user indices before doing this. Right after the `save()` operation above.